flink/flink-1.13/src/main/java/io/tidb/bigdata/flink/telemetry/FlinkTeleMsg.java

```diff
@@ -0,0 +1,233 @@
+
+import static io.tidb.bigdata.flink.connector.table.TiDBDynamicTableFactory.SINK_BUFFER_FLUSH_MAX_ROWS;
+import static io.tidb.bigdata.flink.tidb.TiDBBaseCatalog.TIDB_CATALOG_LOAD_MODE;
+import static io.tidb.bigdata.tidb.ClientConfig.TIDB_FILTER_PUSH_DOWN;
+import static io.tidb.bigdata.tidb.ClientConfig.TIDB_REPLICA_READ;
+import static io.tidb.bigdata.tidb.ClientConfig.TIDB_WRITE_MODE;
+import static org.apache.flink.streaming.connectors.kafka.table.KafkaOptions.SINK_BUFFER_FLUSH_INTERVAL;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.tidb.bigdata.telemetry.TeleMsg;
+import io.tidb.bigdata.tidb.ClientConfig;
+import io.tidb.bigdata.tidb.ClientSession;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.util.EnvironmentInformation;
+import org.apache.flink.util.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tikv.raw.RawKVClient;
+import org.tikv.shade.com.google.protobuf.ByteString;
+
+/** FlinkTeleMsg is a single instance. Only send once in one Flink application. */
+public class FlinkTeleMsg extends TeleMsg {
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  private static volatile FlinkTeleMsg flinkTeleMsg;
+
+  private static final String SUBNAME = "flink-1.13";
+  private static final String TIBIGDATA_FLINK_VERSION = "0.0.5-SNAPSHOT";
+  private static final String TRACK_ID = "TiBigDataFlink1.14TelemetryID";
+  private static final String TRACK_ID_PREFIX = "trkid_";
+  private static final String APP_ID_PREFIX = "appid_";
+  private volatile FlinkTeleMsgState sendState = FlinkTeleMsgState.UNSENT;
+  private Map<String, String> properties;
+
+  private FlinkTeleMsg(Map<String, String> properties) {
+    this.properties = properties;
+    this.instance = setInstance();
+    this.trackId = setTrackId();
+    this.content = setContent();
+    this.subName = setSubName();
+  }
+
+  /**
+   * Get FlinkTeleMsg single instance.
+   *
+   * @param properties tibigdata-flink properties.
+   * @return The flinkTeleMsg is used to be reporting asynchronously.
+   */
+  public static FlinkTeleMsg getInstance(Map<String, String> properties) {
+    if (flinkTeleMsg == null)
+      synchronized (FlinkTeleMsg.class) {
+        if (flinkTeleMsg == null) flinkTeleMsg = new FlinkTeleMsg(properties);
+      }
+    return flinkTeleMsg;
+  }
+
+  /**
+   * Get FlinkTeleMsg single instance. If flinkTeleMsg has been initialized, the object is returned,
+   * otherwise an exception is thrown.
+   *
+   * @return flinkTeleMsg
+   */
+  public static FlinkTeleMsg getInstance() {
```

Comment:
The two getInstance methods have different behavior, better to use different names